src/main/java/org/cbio/gdcpipeline/tasklet/FileMappingTasklet.java

```diff
@@ -0,0 +1,281 @@
+package org.cbio.gdcpipeline.tasklet;
+
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.gdc.nci.tcga.bcr.xml.biospecimen._2.Sample;
+import org.cbio.gdcpipeline.model.gdc.nci.tcga.bcr.xml.biospecimen._2.TcgaBcr;
+import org.cbio.gdcpipeline.model.rest.response.Case;
+import org.cbio.gdcpipeline.model.rest.response.GdcApiResponse;
+import org.cbio.gdcpipeline.model.rest.response.Hits;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+   /*
+     1. Get all TCGA barcodes from biospecimen files alongwith all sample-ids
+        Add barcodes to a list and also to a map with sample_ids
+     2. Use list to call GDC API and create a map for each barcode with filenames
+     3. Add both maps to execution context
+     */
+
+/**
+ * @author Dixit Patel
+ */
+
+public class FileMappingTasklet implements Tasklet {
+
+    private static Log LOG = LogFactory.getLog(FileMappingTasklet.class);
+
+    @Value("#{jobParameters[sourceDirectory]}")
+    private String sourceDir;
+
+    @Value("#{jobParameters[study]}")
+    private String cancer_study_id;
+
+    @Value("${gdc.api.endpoint}")
+    private String GDC_API_ENDPOINT;
+
+    @Value("${gdc.max.response.size}")
+    private int MAX_RESPONSE_SIZE;
+
+    private HashMap<String, List<String>> barcodeToSamplesMap = new HashMap<>();
+    private HashMap<String, List<String>> uuidToFilesMap = new HashMap<>();
+    private RestTemplate restTemplate = new RestTemplate();
+
+
+    protected void xmlUnmarshall(File xmlFile) throws JAXBException {
+
+
+        JAXBContext jaxbContext = JAXBContext.newInstance(TcgaBcr.class);
+        Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+
+        TcgaBcr tcgaBcr = (TcgaBcr) unmarshaller.unmarshal(xmlFile);
+
+        String case_uuid = tcgaBcr.getPatient().getBcrPatientUuid().getValue();
+        case_uuid = case_uuid.toLowerCase();
+        List<String> emptyList = new ArrayList<>();
+        uuidToFilesMap.put(case_uuid, emptyList);
```

Comment:
UUID is needed for API request. Plus UUID is across all studies as per our email conversation with GDC.
But I also create a barcode to sample list Map. 
Added to meeting doc